src/towncrier/newsfragments/608.feature.rst

```diff
@@ -0,0 +1,8 @@
+Issues are now sorted by issue number even if they have non-digit characters.
+For example::
+
+    - some issue (gh-3, gh-10)
+    - another issue (gh-4)
+    - yet another issue (gh-11)
+
+The sorting algorithm groups the issues first by non-text characters and then by number.
```

Comment:
An interesting aside: without some text after the pre-formatted block, it breaks the rendering of the issue number towncrier adds when building because it's considered part of the block.

Reply:
We might want to create a GitHub issue for this with the input, actual result vs expected result

Reply:
Done.